### ISSUE TITLE
Integrate memory for agents

### ISSUE THREAD

**Issue by CodingTux** [nerobossai/nerosdk#4]
Idea is to have
1. Memory of every interactions 
2. Agent can lookup for memory
3. Create interaction graph of multiple users

**Comment by CodeInEcho** [comment]
I suggest developing NeuroSDK as a proxy for general models rather than just conversational models. The memory management design needs to be more detailed. I want to create an AI agent similar to AIXBTA that focuses on knowledge, which requires manual data insertion and integration of externally acquired data. Simply storing conversation memories doesn't seem sufficient to achieve this?

**Comment by CodeInEcho** [comment]
It's recommended to use langchain vector stores for development

**Comment by CodingTux** [comment]
Yeah, also planning to integrate tooling. We need to decide best way forward for configs tho.
